create_per_gene_matrix.py

```diff
@@ -253,69 +308,61 @@ def compute_from_full_mt(chr20: bool, overwrite: bool):
     mt = mt.explode_rows(mt.vep)
     mt = mt.transmute_rows(**mt.vep)
 
-    mt = mt.annotate_cols(
-        pop=['all', mt.meta.pop]
+    mt = mt.annotate_rows(
+                cum_csq=
+                hl.case(missing_false=True)
+                    .when(mt.csq == 0, [CSQ_CODES.index('lof')])
+                    .when(mt.csq == 1, [CSQ_CODES.index('strong_revel_missense'),CSQ_CODES.index('moderate_to_strong_revel_missense'),CSQ_CODES.index('supporting_to_strong_revel_missense'),CSQ_CODES.index('missense')])
+                    .when(mt.csq == 2, [CSQ_CODES.index('moderate_to_strong_revel_missense'),CSQ_CODES.index('supporting_to_strong_revel_missense'),CSQ_CODES.index('missense')])
+                    .when(mt.csq == 3, [CSQ_CODES.index('supporting_to_strong_revel_missense'),CSQ_CODES.index('missense')])
+                    .when(mt.csq == 4, [CSQ_CODES.index('missense')])
+                    .when(mt.csq == 5, [CSQ_CODES.index('synonymous')])
+                    .or_missing()
     )
-    mt = mt.explode_cols(mt.pop)
+
+    mt = mt.explode_rows('cum_csq')
+
+    mt = mt.checkpoint('gs://gnomad-tmp/compound_hets/het_and_hom_per_gene{}.1.mt'.format(
+        '.chr20' if chr20 else ''
+    ), overwrite=True)
 
     mt = mt.group_rows_by(
         'gene_id'
     ).aggregate_rows(
         gene_symbol=hl.agg.take(mt.gene_symbol, 1)[0]
-    ).aggregate(
-        counts=hl.agg.filter(
-            hl.if_else(
-                mt.pop == 'all',
-                hl.is_defined(mt.popmax) & (mt.popmax.AF <= MAX_FREQ),
-                mt.freq[freq_dict[mt.pop]].AF <= MAX_FREQ
-            ),
-            hl.agg.group_by(
-                hl.if_else(
-                    mt.pop == 'all',
-                    mt.popmax.AF > 0.001,
-                    mt.freq[freq_dict[mt.pop]].AF > 0.001
-                ),
-                hl.struct(
-                    hom_csq=hl.agg.filter(~mt.is_het, hl.agg.min(mt.csq)),
-                    het_csq=hl.agg.filter(mt.is_het, hl.agg.min(mt.csq)),
-                    het_het_csq=hl.sorted(
```

Comment:
This I added as Anne had frequently asked me how many individuals are het_het for a certain csq regardless of predicted phase. So its just the number of het_het for the csq without consideration of chet/same_hap/unphased (which are otherwise all counted separately i.e., an individual can be counted once in chet and once in unphased if they have two variant pairs for this csq, one falling in each phase category), so when this is the case, "het_het" would only count this individual once for that csq.

Reply:
In this case the `het_het_csq` is being removed though, is this something we will still want? In the other function there is still an `n_het_het`

Reply:
I didn't look at this closely enough, sorry, yes it can be removed here as it's in the other function.